src/server/location/RoomBookingInfoView.entity.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  ViewEntity, ViewColumn, Connection, SelectQueryBuilder,
+} from 'typeorm';
+import { Room } from './room.entity';
+import { Meeting } from '../meeting/meeting.entity';
+import { DAY, TERM } from '../../common/constants';
+import { CourseInstance } from '../courseInstance/courseinstance.entity';
+import { NonClassEvent } from '../nonClassEvent/nonclassevent.entity';
+import { NonClassParent } from '../nonClassParent/nonclassparent.entity';
+import { Course } from '../course/course.entity';
+import { Semester } from '../semester/semester.entity';
+
+/**
+ * Represents the data associated with a room being booked for a meeting. Can
+ * be used to query whether a room is available at a certain time by the year,
+ * term, day, startTime, and endTime.
+ */
+
+@ViewEntity('RoomBookingInfoView', {
+  expression: (connection: Connection):
+  SelectQueryBuilder<Meeting> => connection.createQueryBuilder()
+    .select('r.id', 'roomId')
+    .addSelect('s.academicYear', 'year')
```

Comment:
ah, that's absolutely right, we do need double quotes there. Thanks for catching that. And actually, I think we want to call this `calendarYear`, because that's what the `academicYear` field in `Semester` actually represents. (We should really fix that...)

I made those changes -- and generated a new migration -- in b013e57e

Reply:
Ah, `calendarYear/academicYear` strikes again! Ok looks good.